src/core/resource.cc

```diff
@@ -259,6 +264,33 @@ io_queue_topology::io_queue_topology(io_queue_topology&& o)
 
 }
 
+std::optional<std::vector<unsigned>>
+resource::internal::cpus_for_numa_node_id(hwloc::internal::topology_holder& topo_holder, unsigned numa_node_id) {
+#ifndef SEASTAR_HAVE_HWLOC
+    return std::nullopt;
+#else
+    auto topo = topo_holder.get();
+    auto numa_node = hwloc_get_obj_by_type(topo, HWLOC_OBJ_NUMANODE, numa_node_id);
+    while (numa_node && numa_node->os_index != numa_node_id) {
+        numa_node = hwloc_get_next_obj_by_type(topo, HWLOC_OBJ_NUMANODE, numa_node);
+    }
+    if (!numa_node) {
+        return std::nullopt; // ??
+    }
+    auto parent = numa_node->parent;
+    auto cpuset = parent->cpuset;
+    cpu_set_t os_cpuset;
+    hwloc_cpuset_to_glibc_sched_affinity(topo, cpuset, &os_cpuset, sizeof(os_cpuset));
+    std::vector<unsigned> ret;
+    for (unsigned idx = 0; idx < CPU_SETSIZE; ++idx) {
+        if (CPU_ISSET(idx, &os_cpuset)) {
```

Comment:
The `cpu_set_t` is converted into `std::vector<int>` here and then back in fa47229a47f6e7564420180c08fa0d9fc5223075. Why not make `cpus_for_numa_node_id` return cpu_set_t right at once? (Or keep it in `resource::resources` as I suggested in above comment

Reply:
There is also a resources::cpuset data type.